dashboard/test/ui/features/step_definitions/ailab_steps.rb

```diff
@@ -0,0 +1,13 @@
+When /^I select table column "([^"]*)"$/ do |column|
+  @elements = @browser.find_elements(:css, ".table-column")
+
+  @element = @elements[column.to_i]
+  @element.click
+end
+
+Then /^I wait to see the dynamic instructions contain "((?:[^"\\]|\\.)*)" key$/ do |selector|
+  steps %{
+    And I wait to see ".#{selector}"
+    And element ".#{selector}" has one of css properties "opacity" equal to "1"
+  }
+end
```

Comment:
The naming here implies that we are waiting until the dynamic instructions contain the given key, but really we are waiting to make sure the key exists and then just making sure it's visible (via opacity) without waiting for that to happen.  Given the way things work on a page with dynamic instructions , I think it might make sense to do a one-off wait for the dynamic instructions to exist at the beginning of a page's test, and then have a step like this which checks that the specific instruction is visible (waiting if necessary, though I'm not sure whether that will be needed or not).

Reply:
To clarify, you're suggesting we remove `And I wait to see ".#{selector}"` from this step and then add it at the beginning of each page? I tested it out by removing it. It caused some errors in seemingly unrelated lines (8 and 60). I fixed those errors by adding `And I wait to see ...` to the beginning of each page. A second fix for it was adding "And I wait for 2 seconds" before those lines that threw errors. I don't think we should add this line to the beginning of each page because adding the same 10 lines is a bit redundant and makes it harder to read. The second fix and leaving it as is both take 36-39 seconds to run.

Would it be okay leave this step as is but change the name to something like, `"I see the dynamic instructions are showing [key]"`?

Reply:
As discussed, once one dynamic instruction selector can be seen in the DOM, they should all be there, but it doesn't actually mean they are visible because we use opacity for that.  That said, that first step can just be done for a single load of AI Lab.  After that, we just need to check the opacity to know whether a dynamic instruction is actually showing.  I'm not sure whether we'll need to wait for that check or not -- if we can possibly get to that step before the opacity has changed then we'd need to wait -- but the step naming should reflect what we do.